File: euler/p2.py
#! /usr/bin/env python

def fib():
    a, b = 1,2
    ans = []
    while True:
        a, b = b, a+b
        if a % 2 == 0:
            ans.append(a)
        if b > 4000000:
            break
    return ans

def main():
    ans = fib()
    sums = 0
    print ans
    while True:
        sums += ans.pop()
        if ans == []:
            break
    print sums

if __name__ == '__main__':
    main()

File: euler/p2.java
public class p2 {
	public static int fib(int n) {
		int ans = 0;
		if (n == 1) ans += 1;
		else if (n == 2) ans += 2;
		else ans += fib(n-1)+fib(n-2);
		return ans;
	}

	public static void main(String[] args) {
		int count = 0;
		for (int i = 0; i < 1000; i++) {
			if (fib(i) > 1000) {
				break;
			}
			count++;
		}
		System.out.println(count);
	}
}
